src/openrct2-ui/windows/TitleMenu.cpp

```diff
@@ -123,178 +63,227 @@ static void WindowTitleMenuScenarioselectCallback(const utf8* path)
+            if (i > 5)
+            {
+                yOffset = -(widget->height() + 5 + (i * 12));
+            }
+
+            WindowDropdownShowText(
+                { windowPos.x + widget->left, windowPos.y + widget->top + yOffset }, widget->height() + 1,
+                TRANSLUCENT(colours[0]), Dropdown::Flag::StayOpen, i);
         }
     }
-#endif
-}
 
-static void WindowTitleMenuDropdown(rct_window* w, WidgetIndex widgetIndex, int32_t dropdownIndex)
-{
-    if (widgetIndex == WIDX_GAME_TOOLS)
+    void OnDropdown(WidgetIndex widgetIndex, int32_t selectedIndex) override
     {
-        switch (dropdownIndex)
+        if (widgetIndex == WIDX_GAME_TOOLS)
         {
-            case DDIDX_SCENARIO_EDITOR:
-                Editor::Load();
-                break;
-            case DDIDX_CONVERT_SAVED_GAME:
-                Editor::ConvertSaveToScenario();
-                break;
-            case DDIDX_TRACK_DESIGNER:
-                Editor::LoadTrackDesigner();
-                break;
-            case DDIDX_TRACK_MANAGER:
-                Editor::LoadTrackManager();
-                break;
-            case DDIDX_OPEN_CONTENT_FOLDER:
+            switch (selectedIndex)
             {
-                auto context = OpenRCT2::GetContext();
-                auto env = context->GetPlatformEnvironment();
-                auto uiContext = context->GetUiContext();
-                uiContext->OpenFolder(env->GetDirectoryPath(OpenRCT2::DIRBASE::USER));
-                break;
+                case DDIDX_SCENARIO_EDITOR:
+                    Editor::Load();
+                    break;
+                case DDIDX_CONVERT_SAVED_GAME:
+                    Editor::ConvertSaveToScenario();
+                    break;
+                case DDIDX_TRACK_DESIGNER:
+                    Editor::LoadTrackDesigner();
+                    break;
+                case DDIDX_TRACK_MANAGER:
+                    Editor::LoadTrackManager();
+                    break;
+                case DDIDX_OPEN_CONTENT_FOLDER:
+                {
+                    auto context = OpenRCT2::GetContext();
+                    auto env = context->GetPlatformEnvironment();
+                    auto uiContext = context->GetUiContext();
+                    uiContext->OpenFolder(env->GetDirectoryPath(OpenRCT2::DIRBASE::USER));
+                    break;
+                }
+                default:
+                    InvokeCustomToolboxMenuItem(selectedIndex - DDIDX_CUSTOM_BEGIN);
+                    break;
             }
-            default:
-                InvokeCustomToolboxMenuItem(dropdownIndex - DDIDX_CUSTOM_BEGIN);
-                break;
         }
     }
-}
 
-static void WindowTitleMenuCursor(
-    rct_window* w, WidgetIndex widgetIndex, const ScreenCoordsXY& screenCoords, CursorID* cursorId)
-{
-    gTooltipTimeout = 2000;
-}
+    CursorID OnCursor(WidgetIndex widgetIndex, const ScreenCoordsXY& screenCoords, CursorID cursorId) override
```

Comment:
```suggestion
    CursorID OnCursor(WidgetIndex, const ScreenCoordsXY&, CursorID cursorId) override
```
Unused parameters may be omitted.